crates/ln-dlc-node/src/node/payment_persister.rs

```diff
@@ -0,0 +1,92 @@
+use crate::HTLCStatus;
+use crate::MillisatAmount;
+use crate::PaymentFlow;
+use crate::PaymentInfo;
+use anyhow::Result;
+use lightning::ln::PaymentHash;
+use lightning::ln::PaymentPreimage;
+use lightning::ln::PaymentSecret;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::sync::Mutex;
+use time::OffsetDateTime;
+
+pub trait PaymentPersister: Clone {
+    fn insert(&self, payment_hash: PaymentHash, info: PaymentInfo) -> Result<()>;
+    fn upsert(
+        &self,
+        payment_hash: &PaymentHash,
+        flow: PaymentFlow,
+        htlc_status: HTLCStatus,
+        preimage: Option<PaymentPreimage>,
+        secret: Option<PaymentSecret>,
+    ) -> Result<()>;
+    fn get(&self, payment_hash: &PaymentHash) -> Result<Option<(PaymentHash, PaymentInfo)>>;
+    fn all(&self) -> Result<Vec<(PaymentHash, PaymentInfo)>>;
+}
+
+#[derive(Default, Clone)]
+pub struct PaymentMap(Arc<Mutex<HashMap<PaymentHash, PaymentInfo>>>);
+
+impl PaymentPersister for PaymentMap {
+    fn insert(&self, payment_hash: PaymentHash, info: PaymentInfo) -> Result<()> {
+        self.0.lock().unwrap().insert(payment_hash, info);
+
+        Ok(())
+    }
+
+    fn upsert(
+        &self,
+        payment_hash: &PaymentHash,
+        flow: PaymentFlow,
+        htlc_status: HTLCStatus,
+        preimage: Option<PaymentPreimage>,
+        secret: Option<PaymentSecret>,
+    ) -> Result<()> {
+        let mut payments = self.0.lock().unwrap();
```

Comment:
did you mean to keep this as `unwrap()`? this seems to be the only opportunity to get an error in this call

Reply:
I did mean to keep it, because I think `unwrap`ping `LockResult`s is usually best.

The reason why we have to return an error is because we are implementing a trait method which returns a `Result`. For a `HashMap` it's irrelevant, but when accessing a database it's a different story.